dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/DotReactiveElasticsearchClient.java

```diff
@@ -0,0 +1,874 @@
+    public Flux<SearchHit> scroll(HttpHeaders headers, SearchRequest searchRequest) {
+
+        TimeValue scrollTimeout = searchRequest.scroll() != null ? searchRequest.scroll().keepAlive()
+                : TimeValue.timeValueMinutes(1);
+
+        if (searchRequest.scroll() == null) {
+            searchRequest.scroll(scrollTimeout);
+        }
+
+        EmitterProcessor<ActionRequest> outbound = EmitterProcessor.create(false);
+        FluxSink<ActionRequest> request = outbound.sink();
+
+        EmitterProcessor<SearchResponse> inbound = EmitterProcessor.create(false);
+
+        Flux<SearchResponse> exchange = outbound.startWith(searchRequest).flatMap(it -> {
+
+            if (it instanceof SearchRequest) {
+                return sendRequest((SearchRequest) it, DotReactiveElasticsearchClient.RequestCreator
+                        .search(), SearchResponse.class, headers);
+            } else if (it instanceof SearchScrollRequest) {
+                return sendRequest((SearchScrollRequest) it, DotReactiveElasticsearchClient.RequestCreator
+                        .scroll(), SearchResponse.class, headers);
+            } else if (it instanceof ClearScrollRequest) {
+                return sendRequest((ClearScrollRequest) it, DotReactiveElasticsearchClient.RequestCreator
+                        .clearScroll(), ClearScrollResponse.class, headers)
+                        .flatMap(discard -> Flux.empty());
+            }
+
+            throw new IllegalArgumentException(
+                    String.format("Cannot handle '%s'. Please make sure to use a 'SearchRequest' or 'SearchScrollRequest'.", it));
+        });
+
+        return Flux.usingWhen(Mono.fromSupplier(DotReactiveElasticsearchClient.ScrollState::new),
+
+                scrollState -> {
+
+                    Flux<SearchHit> searchHits = inbound.<SearchResponse> handle((searchResponse, sink) -> {
+
+                        scrollState.updateScrollId(searchResponse.getScrollId());
+                        if (isEmpty(searchResponse.getHits())) {
+
+                            inbound.onComplete();
+                            outbound.onComplete();
+
+                        } else {
+
+                            sink.next(searchResponse);
+
+                            SearchScrollRequest searchScrollRequest = new SearchScrollRequest(scrollState.getScrollId())
+                                    .scroll(scrollTimeout);
+                            request.next(searchScrollRequest);
+                        }
+
+                    }).map(SearchResponse::getHits) //
+                            .flatMap(Flux::fromIterable);
+
+                    return searchHits.doOnSubscribe(ignore -> exchange.subscribe(inbound));
+
+                }, state -> cleanupScroll(headers, state), //
+                state -> cleanupScroll(headers, state), //
+                state -> cleanupScroll(headers, state)); //
+    }
+
+    private static boolean isEmpty(@Nullable SearchHits hits) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'hits' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5113278)